clients/src/main/java/org/apache/kafka/clients/consumer/SubscriptionPattern.java

```diff
@@ -0,0 +1,28 @@
+/*
+ * Licensed to the Apache Software Foundation (ASF) under one or more
+ * contributor license agreements. See the NOTICE file distributed with
+ * this work for additional information regarding copyright ownership.
+ * The ASF licenses this file to You under the Apache License, Version 2.0
+ * (the "License"); you may not use this file except in compliance with
+ * the License. You may obtain a copy of the License at
+ *
+ *    http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+package org.apache.kafka.clients.consumer;
+
+public class SubscriptionPattern {
```

Comment:
This class would need some doc explaining what it represents. Also not sure if this is the right place for it (given that this whole intention with the new regex is driven by the broker, but is not implemented yet). So at this point is not clear to me if we would prefer to define this on the broker side to be used there? Could be. As suggested in the Jira, maybe we should wait for the broker implementation of the new regex, and then align on this class?

Reply:
I think we should leave it on the consumer side, as I think its functionalities is more to differenciate between using Java.util.Pattern and using regex that compatible with RE2J. If we were just to use a string instead of SubscribedPattern, the user may get confused about why they need to use Pattern for the other subsribe methods.

Reply:
> This class would need some doc explaining what it represents.

I definitely agree with @lianetm on this.

> Also not sure if this is the right place for it (given that this whole intention with the new regex is driven by the broker, but is not implemented yet). So at this point is not clear to me if we would prefer to define this on the broker side to be used there?

The KIP says this class should be added to the public API of the consumer. I agree with @Phuc-Hong-Tran that this class is merely there to differentiate patterns that are Google RE2/J from `java.util.regex.Pattern` as stated in the KIP.